pkg/controller/inferenceservice/reconcilers/knative/service_reconciler.go

```diff
@@ -44,13 +45,15 @@ type ServiceReconciler struct {
 	client         client.Client
 	scheme         *runtime.Scheme
 	serviceBuilder *knative.ServiceBuilder
+	recorder       record.EventRecorder
```

Comment:
I do not see where recorder is instantiated in real controller. I saw in test it is actually created by `recorder := mgr.GetEventRecorderFor(fmt.Sprintf("ServiceReconciler"))`

Reply:
look at: https://github.com/kubeflow/kfserving/pull/876/files#diff-6d0b9f95ed39eac3ea82e113afd8500bR148